eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqlitePaymentsDb.scala

```diff
@@ -155,8 +155,9 @@ class SqlitePaymentsDb(sqlite: Connection) extends PaymentsDb with Logging {
   }
 
   override def getPendingPaymentRequestAndPreimage(paymentHash: ByteVector32): Option[(ByteVector32, PaymentRequest)] = {
-    using(sqlite.prepareStatement("SELECT payment_request, preimage FROM received_payments WHERE payment_hash = ? AND received_at IS NULL")) { statement =>
+    using(sqlite.prepareStatement("SELECT payment_request, preimage FROM received_payments WHERE payment_hash = ? AND received_at IS NULL AND (expire_at > ? OR expire_at IS NULL)")) { statement =>
       statement.setBytes(1, paymentHash.toArray)
+      statement.setLong(2, Platform.currentTime)
```

Comment:
ok because we store `expire_at` as milliseconds:

https://github.com/ACINQ/eclair/blob/667444cd0eb8917e2cf6e8f10773d46fb7f0f742/eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqlitePaymentsDb.scala#L140

Reply:
I don't like having non-deterministic db calls, why not check for expired PR at the call site instead ?

Reply:
The `expire_at` db column was meant to provide a database level filter for non expired invoice, in this way we don't have to retrieve all of them and apply the filter later.

Reply:
Sorry but I really think it's the wrong place and also makes testing more complex, I would like to see an explicit check in the upper layer